idb/postgres/postgres_migrations.go

```diff
@@ -790,97 +686,372 @@ func m5RewardsAndDatesPart2UpdateAccounts(db *IndexerDb, state *MigrationState,
+			return nil, fmt.Errorf("m7: unable to scan a row: %v", err)
 		}
 
-		// 4. setCreateCloseAssetHolding - (upsert) set the accounts asset holding create/close rounds.
-		err = executeForEachCreatable(setCreateCloseAssetHolding, address[:], result.assetHolding)
+		var stxn types.SignedTxnWithAD
+		err = msgpack.Decode(txnBytes, &stxn)
 		if err != nil {
-			return fmt.Errorf("%s: failed to update %s with asset holding create/close: %v", rewardsCreateCloseUpdateErr, addressStr, err)
+			return nil, fmt.Errorf("m7: unable to scan a row: %v", err)
+		}
+
+		participants := getParticipants(stxn)
+		for _, address := range participants {
+			// Don't update special accounts (m10 fixes this).
+			if (address != specialAccounts.RewardsPool) && (address != specialAccounts.FeeSink) {
+				txnID := txnID{round, intra}
+
+				storedTxnID, ok := res[address]
+				if !ok {
+					res[address] = txnID
+				} else {
+					if txnIDLess(txnID, storedTxnID) {
+						res[address] = txnID
+					}
+				}
+			}
+		}
+
+		numRows++
+		if numRows%1000000 == 0 {
+			db.log.Printf("read %d transactions (pass 1)", numRows)
+		}
+	}
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("m7: error scanning rows: %v", err)
+	}
+	db.log.Print("finished reading transactions (pass 1)")
+
+	return res, nil
+}
+
+func getAccountsWithoutTxnData(db *IndexerDb, specialAccounts idb.SpecialAccounts, accountsFirstUsed map[sdk_types.Address]txnID) ([]addressAccountData, error) {
+	// Set Created, Deleted for accounts with no transactions.
+	// Genesis accounts could have this property.
+	// Query accounts.
+	res := []addressAccountData{}
+
+	options := idb.AccountQueryOptions{}
+	options.IncludeDeleted = true
+
+	db.log.Print("querying accounts")
+	accountCh, _ := db.GetAccounts(context.Background(), options)
+
+	// Read all accounts.
+	numRows := 0
+	db.log.Print("started reading accounts")
+	for accountRow := range accountCh {
+		if accountRow.Error != nil {
+			return nil, fmt.Errorf("m7: problem querying accounts: %v", accountRow.Error)
 		}
 
-		// 5. setCreateCloseApp          - set the accounts created apps create/close rounds.
-		err = executeForEachCreatable(setCreateCloseApp, address[:], result.app)
+		address, err := sdk_types.DecodeAddress(accountRow.Account.Address)
 		if err != nil {
-			return fmt.Errorf("%s: failed to update %s with app create/close: %v", rewardsCreateCloseUpdateErr, addressStr, err)
+			return nil, fmt.Errorf("m7: failed to decode address %s err: %v",
+				accountRow.Account.Address, err)
 		}
 
-		// 6. setCreateCloseAppLocal     - (upsert) set the accounts local apps create/close rounds.
-		err = executeForEachCreatable(setCreateCloseAppLocal, address[:], result.appLocal)
+		// Don't update special accounts (m10 fixes this)
+		if (address != specialAccounts.FeeSink) && (address != specialAccounts.RewardsPool) {
+			if _, ok := accountsFirstUsed[address]; !ok {
```

Comment:
We should also check if the account was created after maxRound. I believe the deleted flag wouldn't be reset, but we set created_at unconditionally so it would overwrite create_at to 0 if it was created after maxRound.

Maybe I'm missing something, I would have expected the previous tests to have hit this problem.

Reply:
Good catch. Maybe we were just lucky. Pass 1 over transactions and fetching accounts takes several minutes for me.